ghcide/session-loader/Development/IDE/Session.hs

```diff
@@ -266,7 +266,7 @@ loadWithImplicitCradle mHieYaml rootDir = do
 
 getInitialGhcLibDirDefault :: Recorder (WithPriority Log) -> FilePath -> IO (Maybe LibDir)
 getInitialGhcLibDirDefault recorder rootDir = do
-  hieYaml <- findCradle def rootDir
+  hieYaml <- findCradle def (rootDir </> "a")
```

Comment:
these changes are debugging code I assume?

Reply:
not quite, we do have these kind of magic constants all over the place:
* https://github.com/haskell/haskell-language-server/blob/master/exe/Wrapper.hs#L224
* https://github.com/haskell/haskell-language-server/blob/master/src/Ide/Main.hs#L108

The reason is `findCradle` only works for Filepaths, not directories. For finding the cradle type, we don't need a real Filepath, which is why we have these constants.

However, this seems to be a missing feature in `hie-bios`'s API, not a feature.
So, if it speeds up CI, then I would merge this, update hie-bios's API, and then undo these magic filepaths.
But I am unfortunately not sure, they truly speed up CI.

Reply:
Oh I see. Well then they need comments!